uefi/src/proto/debug/mod.rs

```diff
@@ -170,3 +170,46 @@ pub enum ProcessorArch: u32 => {
     /// RISC-V 128-bit
     RISCV_128   = 0x5128,
 }}
+
+/// The debug port protocol abstracts the underlying debug port
+/// hardware, whether it is a regular Serial port or something else.
+#[repr(C)]
+#[unsafe_protocol("eba4e8d2-3858-41ec-a281-2647ba9660d0")]
+pub struct DebugPort {
+    reset: extern "efiapi" fn(this: &DebugPort) -> Status,
+    write: extern "efiapi" fn(
+        this: &DebugPort,
+        timeout: u32,
+        buffer_size: &mut usize,
+        buffer: *const c_void,
+    ) -> Status,
+    read: extern "efiapi" fn(
+        this: &DebugPort,
+        timeout: u32,
+        buffer_size: &mut usize,
+        buffer: *mut c_void,
+    ) -> Status,
+    poll: extern "efiapi" fn(this: &DebugPort) -> Status,
+}
+
+impl DebugPort {
+    /// Flushes the port.
+    pub fn reset(&self) -> Result {
+        (self.reset)(self).into()
+    }
+
+    /// Writes 8 bytes at a time to the underlying SerialIo protocol.
```

Comment:
Some suggestions for the docstring (and similar for `read`):
* It's not 8 bytes at a time, it's `buffer_size` bytes (but see below about making it a slice)
* It's not necessarily SerialIo right? It's a "debugport device"
* Should mention that the timeout unit is microseconds